Add missing dependency to //tensorflow/compiler/tests:xla_test for ROCm

wenchenvincent opened the pull request:
The following tests needs tensorflow.python.platform.sysconfig //tensorflow/compiler/tests:matrix_solve_op_test_cpu //tensorflow/compiler/tests:matrix_solve_op_test_cpu_mlir_bridge_test

wenchenvincent committed:
[ROCm] Only enable xla_gpu_enable_cublaslt for CUDA on matrix_solve_op_test

BrianWieder commented:
Could you point me to where //tensorflow/compiler/tests:matrix_solve_op_test_cpu and //tensorflow/compiler/tests:matrix_solve_op_test_cpu_mlir_bridge_test need/import tensorflow.python.platform.sysconfig?

From https://github.com/tensorflow/tensorflow/blob/master/tensorflow/compiler/tests/matrix_solve_op_test.py I don't see any imports of sysconfig

wenchenvincent commented:
> Could you point me to where //tensorflow/compiler/tests:matrix_solve_op_test_cpu and //tensorflow/compiler/tests:matrix_solve_op_test_cpu_mlir_bridge_test need/import tensorflow.python.platform.sysconfig?
> 
> From https://github.com/tensorflow/tensorflow/blob/master/tensorflow/compiler/tests/matrix_solve_op_test.py I don't see any imports of sysconfig

Sorry for the confusion. The context was that the recent upstream refactoring made the //tensorflow/compiler/tests:matrix_solve_op_test_cpu test in our ROCm fork fail (https://github.com/ROCmSoftwarePlatform/tensorflow-upstream) due to missing dependency. So I added the fix and would like to upstream. However, I didn't realize that the previous change in this test to disable cublasLt was never upstreamed. Now I have just added that commit in this PR as well.

BrianWieder commented:
Thanks for the context. Now that this is modifying XLA flags, I don't think that I am the right reviewer. @gbaned Could you find another reviewer for this PR?

reedwm commented on tensorflow/compiler/tests/BUILD:
```diff
@@ -56,6 +56,7 @@ py_library(
         "//tensorflow/python/compiler/xla:compiler_py",
         "//tensorflow/python/platform:flags",
         "//tensorflow/python/platform:tf_logging",
+        "//tensorflow/python/platform:sysconfig",
```
Why not add the dep to the `matrix_solve_op_test` build rule directly?

wenchenvincent replied:
There is another PR also needing this dependency (https://github.com/tensorflow/tensorflow/pull/60696). The common dependency shared by these two tests for ROCm would be xla_test. And we would anticipate that there might be more xla_test that might need to differentiate between ROCm and CUDA. So instead of adding it to the dependencies of these two tests respectively, adding it to the dependencies of xla_test would be a bit more future proof.

akuegel replied:
Each test that uses the import line for sysconfig will need the dependency directly, not indirectly though the xla_test target. This is enforced by the strict deps check.

wenchenvincent replied:
Thanks for pointing it out. I have updated the PR with the change.

gbaned commented:
Hi @wenchenvincent Can you please take look on below internal errors. Thank you!

ERROR: Strict deps violations: //tensorflow/compiler/tests:matrix_solve_op_test_cpu
  In third_party/tensorflow/compiler/tests/matrix_solve_op_test.py, no direct deps found for imports:
    line 26: from tensorflow.python.platform import sysconfig: Module "tensorflow.python.platform" not provided by a direct dep


Use --define=PYDEPS=<OFF|WARN> to temporarily change how strict deps are enforced.
Bug? http://go/py-strict-deps-bug

*** Please fix target dependencies. ***

build_cleaner //third_party/tensorflow/compiler/tests:matrix_solve_op_test_cpu

reedwm reviewed (changes requested):
Can you resolve conflicts?
